subprojects/test-runner/device-provider/impl/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/ReservationDeploymentFactoryImpl.kt

```diff
@@ -136,7 +137,7 @@ internal class ReservationDeploymentFactoryImpl(
             .plus("deploymentName" to deploymentName)
 
         return newDeployment {
-            apiVersion = "extensions/v1beta1"
+            apiVersion = if (useLegacyExtensionsV1Beta) "extensions/v1beta1" else "apps/v1"
```

Comment:
Probably out of the scope of this PR, but it might be better to express a contract in terms of k8s versions because it already defines API. For the client might be easier to think and operate in this way.

This is how I see an ideal solution. Test runner checks k8s version automatically and does fail-fast check: 
"The test runner is supporting only N...M k8s version, but pointed to X version".

If this is not reasonable, we could move this knowledge and responsibility to the client configuration. It may look like a leaky abstraction but will help in troubleshooting.

Example use case to illustrate the problem:

I pointed the test runner to the newest k8s version and faced other backward compatibility problems.
How should I know what version is supported?

Reply:
i think we should express somewhere in documentation that only `apps/v1` versions are supported
This flag will be remove alongside with the `extensions/v1beta` support.

More extendible client with multiple supported API versions maybe a good idea, but there is only `apps/v1' in the wild for now afaik

Reply:
> express somewhere in documentation that only apps/v1 versions are supported

The idea is that this is harder to find. E.g., I want to run tests against GCE. It'd easier to check k8s version than digging into API details.

Reply:
i hope we can skip this problem entirely until `apps/v2` will be released
you can't create something as old as 1.8 in GCE right now

![image](https://user-images.githubusercontent.com/1105133/126154968-61c6a8fe-4411-4eef-8c36-a58edca5ebbc.png)